tools/fragpipe/macros.xml

```diff
@@ -0,0 +1,1206 @@
+<macros>
+    <import>msfragger_macros.xml</import>
+    <token name="@TOOL_VERSION@">20.0</token>
+    <token name="@VERSION_SUFFIX@">0</token>
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@TOOL_VERSION@">fragpipe</requirement>
+        </requirements>
+    </xml>
+
+    <!-- TMT-n uses the following n first labels from TMT-18 -->
+    <token name="@TMT_LABELS@">['126', '127N', '127C', '128N', '128C', '129N', '129C', '130N', '130C', '131N', '131C', '132N', '132C', '133N', '133C', '134N', '134C', '135N']</token>
+    <token name="@iTRAQ-4_LABELS@">['114', '115', '116', '117']</token>
+    <token name="@iTRAQ-8_LABELS@">['113', '114', '115', '116', '117', '118', '119', '121']</token>
+
+    <!--
+      User specifies input file format and selects sample files as are included by the scan_inputs macro
+
+      User must also supply their own manifest file, a TSV file of the format:
+      name, experiment, bioreplicate, data type
+    -->
+    <xml name="samples">
+        <param name="inputs" type="data" format="mzML,mzXML" multiple="true" label="Proteomics Spectrum files"  help="All input scan files must of a matching format: mzML, mzXML"/>
+        <param name="input_prefix" type="text" value="" optional="true" label="File name prefix" help="Names inputs: prefix_rep#.mzXML Leave blank to use History names of inputs">
+              <validator type="regex" message="">[a-zA-Z][a-zA-Z0-9_-]*</validator>
```

Comment:
Interesting that the validator does not give an error when the value is `""`. But makes sense for optional parameter.

Reply:
I suppose the assumption is that the author will ensure the default value is acceptable. Sounds like this is fine as is.